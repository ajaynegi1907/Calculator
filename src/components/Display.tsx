
export interface DisplayProps {
  expression: string;
  result?: number;
}

const Display = ({ expression, result }: DisplayProps): JSX.Element => {
  return (
    <div className="display">
      <div className="display-expression">{expression}</div>
      <div className="display-result">{result !== undefined ? result : " "}</div>
    </div>
  );
};

export default Display;
