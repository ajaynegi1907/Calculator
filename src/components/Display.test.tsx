import { shallow } from "enzyme";
import Display, { DisplayProps } from "./Display";

const props: DisplayProps = {
    expression: "4+5",
    result: 9
};

it("should match snapshot", () => {
    const wrapper  = shallow(<Display {...props} />);
    expect(wrapper).toMatchSnapshot();
})

it("should render empty string as result when result is undefined", () => {
    const customProps: DisplayProps = {
        expression: "4 + 5"
    }

    const wrapper = shallow(<Display {...customProps} />);
    const resultDivContent = wrapper.find(".display-result").getElement().props.children;

    expect(resultDivContent).toEqual(" ");
});