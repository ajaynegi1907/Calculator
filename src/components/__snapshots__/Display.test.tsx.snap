// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="display"
>
  <div
    className="display-expression"
  >
    4+5
  </div>
  <div
    className="display-result"
  >
    9
  </div>
</div>
`;
